services/controller/src/plz/controller/main.py

```diff
@@ -224,6 +229,14 @@ def get_output_files_entrypoint(execution_id):
     return Response(response, mimetype='application/octet-stream')
 
 
+def get_metadata(execution_id):
+    with results_storage.get(execution_id) as results:
+        if results is None:
+            raise ExecutionIdNotFound(
+                f'Metadata not found for execution id: {execution_id}')
+        return ''.join(str(r, 'utf-8') for r in results.metadata())
```

Comment:
I think `r.encode('utf-8')` is more standard here.

Reply:
I've seen a discussion somewhere. There was the argument that using `str` is more pythonic, and I kinda agree.